src/core/components/transform.rs

```diff
@@ -55,79 +56,79 @@ impl std::ops::Add for Transform {
     }
 }
 
-#[derive(Clone, Copy, Debug, DeJson)]
-pub struct Scale {
-    pub x: f32,
-    pub y: f32,
-}
+#[derive(Clone, Copy, Debug)]
+pub struct Scale(pub Vec2);
+
 impl Scale {
     pub fn new(x: f32, y: f32) -> Self {
-        Scale { x, y }
+        Self(vec2(x, y))
     }
 }
 impl Default for Scale {
     fn default() -> Self {
-        Scale::new(0.0, 0.0)
+        Scale::new(1.0, 1.0)
     }
 }
 impl std::ops::Sub for Scale {
     type Output = Self;
 
     fn sub(self, other: Self) -> Self::Output {
-        Self {
-            x: self.x - other.x,
-            y: self.y - other.y,
-        }
+        Self(self.0 - other.0)
     }
 }
 impl std::ops::Add for Scale {
     type Output = Self;
 
     fn add(self, other: Self) -> Self::Output {
-        Self {
-            x: self.x + other.x,
-            y: self.y + other.y,
-        }
+        Self(self.0 + other.0)
     }
 }
 
-#[derive(Clone, Copy, Debug, DeJson)]
-pub struct Translation {
-    pub x: f32,
-    pub y: f32,
-}
-impl Translation {
-    pub fn new(x: f32, y: f32) -> Self {
-        Translation { x, y }
-    }
+pub type Translation = Vec2;
```

Comment:
I think we should keep Translation as a new separate type, it allows us to scale it up with more fields/functionality in the future if we ever want to. Additionally we don't have to have separate structs designated for de/serialization -> Translation